File: app/src/main/java/org/nitri/opentopo/domain/DistancePoint.java
package org.nitri.opentopo.domain;

/**
 * GPX point on distance line
 *
 * Note: io.ticofab.androidgpxparser.parser.domain.Point is package private
 */
public class DistancePoint {

    private final Double mDistance;
    private final Double mElevation;

    DistancePoint(Builder builder) {
        mDistance = builder.mDistance;
        mElevation = builder.mElevation;
    }

    /**
     * @return the distance in meters
     */
    public Double getDistance() {
        return mDistance;
    }

    /**
     * @return the elevation in meters
     */
    public Double getElevation() {
        return mElevation;
    }

    public static class Builder {

        private Double mDistance;
        private Double mElevation;

        public Builder setDistance(Double distance) {
            mDistance = distance;
            return this;
        }

        public Builder setElevation(Double elevation) {
            mElevation = elevation;
            return this;
        }

        public DistancePoint build() {
            return new DistancePoint(this);
        }
    }

}

File: app/src/main/java/org/nitri/opentopo/domain/DistancePoint.kt
package org.nitri.opentopo.domain

/**
 * GPX point on distance line
 *
 * Note: io.ticofab.androidgpxparser.parser.domain.Point is package private
 */
class DistancePoint internal constructor(builder: Builder) {
    /**
     * @return the distance in meters
     */
    val distance: Double?

    /**
     * @return the elevation in meters
     */
    val elevation: Double?

    init {
        distance = builder.mDistance
        elevation = builder.mElevation
    }

    class Builder {
        var mDistance: Double? = null
        var mElevation: Double? = null
        fun setDistance(distance: Double?): Builder {
            mDistance = distance
            return this
        }

        fun setElevation(elevation: Double?): Builder {
            mElevation = elevation
            return this
        }

        fun build(): DistancePoint {
            return DistancePoint(this)
        }
    }
}
